plugin-server/src/types.ts

```diff
@@ -531,22 +552,29 @@ export interface Event {
     group4_properties: Record<string, any>
 }
 
-export interface ClickHouseEvent extends Omit<Event, 'id' | 'elements' | 'elements_hash'> {
-    uuid: string
-    elements_chain: string | undefined
+/** Event in a database-agnostic shape, AKA an ingestion event.
+ * This is what should be passed around most of the time in the plugin server.
+ */
+interface BaseIngestionEvent {
+    eventUuid: string
+    event: string
+    ip: string | null
+    teamId: TeamId
+    distinctId: string
+    properties: Properties
+    timestamp: DateTime
+    elementsList: Element[]
 }
 
-// Clickhouse event as read from kafka
-export interface ClickhouseEventKafka {
-    event: string
+/** Ingestion event before saving, currently just an alias of BaseIngestionEvent. */
+export type PreIngestionEvent = BaseIngestionEvent
+
+/** Ingestion event after saving, currently just an alias of BaseIngestionEvent */
+export type PostIngestionEvent = BaseIngestionEvent
+
+/** Ingestion event that can be cloned by Piscina - only for communication between threads. */
+export interface ClonableIngestionEvent extends Omit<PostIngestionEvent, 'timestamp'> {
```

Comment:
IMO this type existing indicates a self-introduced problem with the new types - we're now parsing-serializing-parsing-serializing. We'll
- forget to do that in some cases, leading to bugs. See e.g. result from `runEventPipeline`
- not test this properly (as elsewhere in this PR)